wordpress-mobile/WordPress-iOS
# Reblog button in ReaderPostCardCell and ReaderDetailViewController

## This PR adds the reblog button in ReaderPostCardCell and ReaderDetailViewController

Ref: #12960

## Description
The goal of this PR is to add the reblog button in the reader, both in `ReaderPostCardCell` and in `ReaderDetailViewController`. 

The design specs have been applied (see extract from the design spec for `ReaderDetailViewController`)
![reblog-action-location_option2](https://user-images.githubusercontent.com/34376330/70409488-8963b280-1a11-11ea-8716-10807cc95aea.png)

in `ReaderPostCardCel`l, the reblog button is placed between the save and the comment buttons.

Accessibility support has been added.

Unit tests are included (extended `ReaderPostCardCellTests.swift` and created `ReaderDetailViewControllerTests.swift`).

#### No action is associated to the reblog button yet.
#### Analytics (tracks) are yet to be added.

##To test
- Build and run the app
- Tap on reader in the tab bar
- Navigate to one of your sites
- Find a post that supports actions
- Check that the reblog button is between the save and comment buttons
- Check that the reblog button highlights on tap (no action will be performed at this moment)
- Navigate to the detail view by tapping on the image or text of the post
- Check that the reblog button is present, as shown in the picture above
- Check that the reblog button highlights on tap (no action will be performed at this moment)
- Switch to dark mode and make sure that the reblog button has consistent appearance with the other buttons

##Notes
- Buttons have been added in the existing `StackViews` that contained the original action buttons
- The associated class is `PostMetaButton`
- Two methods have been added in `WPStyleGuide+Reader.swift`:
   - `applyReaderReblogActionButtonStyle(_ : , _ : )`: overall configuration for the reblog button
   - `applyReaderReblogActionButtonTitle(_ : , - : )`: label configuration for the reblog, allows to define an empty title (for the scenarios where we drop the labels)
  - The above methods are used in both `ReaderPostCardCell` and `ReaderDetailViewController` 
- Still in `WPStyleGuide+Reader.swift`,  `applyReaderSaveForLaterButtonTitles(_ : , _ : )` has been slightly modified to support empty title and it's used in both `ReaderPostCardCell` and `ReaderDetailViewController`
